core/src/main/kotlin/api/ProductsRoute.kt

```diff
@@ -423,17 +423,27 @@ fun Route.products() = route("products/{productId}") {
                 return@post
             }
 
-            val repositoryIds = if (createOrtRun.repositoryIds.isEmpty()) {
-                productService.getRepositoryIdsForProduct(productId)
-            } else {
+            val repositoryIds = if (createOrtRun.repositoryIds.isNotEmpty()) {
                 val productRepoIds = productService.getRepositoryIdsForProduct(productId)
                 require(createOrtRun.repositoryIds.all { it in productRepoIds }) {
                     """
                     The following repository IDs do not belong to product $productId: 
                     ${createOrtRun.repositoryIds.filter { it !in productRepoIds }}
                     """.trimIndent()
                 }
+
                 createOrtRun.repositoryIds
+            } else if (createOrtRun.repositoryFailedIds.isNotEmpty()) {
+                val repositoryIds = productService.getLatestOrtRunWithFailedStatusForProduct(productId)
+                require(createOrtRun.repositoryFailedIds.all { it in repositoryIds }) {
```

Comment:
I am not to sure about the error handling. Ideally you would want two separate kind of errors:
1. Fail if some repository ids that don't belong to the product have been given
2. Fail if some repository ids which _latest_ runs are not in failed state (see underneath)  have been given